### PR TITLE
Layout tests: Disable `implicitly-unsigned-literal` warning

### PR DESCRIPTION
**Pull request by ehaas**
This was causing passing tests to show up as failed.

### CODE REVIEW

**Comment by Vexu** [comment]
IMO this is a bug in the generated test files and should be fixed there instead.

I don't know how to fix it in the generator but locally it can be fixed with this patch:
```diff
diff --git a/test/records/0062_test.c b/test/records/0062_test.c
index 2e3df69..f100b88 100644
--- a/test/records/0062_test.c
+++ b/test/records/0062_test.c
@@ -1606,7 +1606,7 @@ struct A075_extra_size {
 struct A075_extra_size var320;
 
 typedef enum {
-    F321 = 9223372036854775808,
+    F321 = 9223372036854775808ULL,
 } __attribute__((packed)) A076;
 A076 var322;
 struct A076_extra_alignment {
@@ -3206,7 +3206,7 @@ struct A155_extra_size {
 struct A155_extra_size var640;
 
 typedef enum {
-    F641 = -9223372036854775808,
+    F641 = -9223372036854775808ULL,
 } __attribute__((packed)) A156;
 A156 var642;
 struct A156_extra_alignment {
@@ -4831,7 +4831,7 @@ struct B075_extra_size {
 struct B075_extra_size var965;
 
 typedef enum {
-    F966 = 9223372036854775808,
+    F966 = 9223372036854775808ULL,
 } B076;
 B076 var967;
 struct B076_extra_alignment {
@@ -6431,7 +6431,7 @@ struct B155_extra_size {
 struct B155_extra_size var1285;
 
 typedef enum {
-    F1286 = -9223372036854775808,
+    F1286 = -9223372036854775808ULL,
 } B156;
 B156 var1287;
 struct B156_extra_alignment {
```

**Comment by ehaas** [comment]
Made the change & sent a PR upstream to the test generator repo.

I also rolled in a few other quick fixes for arm double/long double alignment, and zero-width bitfield alignment.

**Comment by Vexu** [comment]
> Made the change & sent a PR upstream to the test generator repo.

Nice, thanks!

File: src/record_layout.zig [hunk 1]
```diff
@@ -116,18 +116,23 @@ const SysVContext = struct {
...
-        // Some targets ignore the alignment of the underlying type when laying out
-        // non-zero-sized bit-fields. See test case 0072. On such targets, bit-fields never
-        // cross a storage boundary. See test case 0081.
-        if (comp.ignoreNonZeroSizedBitfieldTypeAlignment()) {
-            ty_fld_algn_bits = 1;
-        }
-
-        if (bit_width == 0) {
+        if (bit_width > 0) {
+            std.debug.assert(bit_width <= ty_size_bits); // TODO: currently checked in parser, move check here?
```
Comment (Vexu) [inline src/record_layout.zig:123]: Why would this check be moved?

Comment (ehaas) [reply src/record_layout.zig:123]: I meant to ask that in a comment on here; we can remove the todo and just leave the `checked in parser` part.

**Pull request merged**